### PR TITLE
allow querying trip by name

### PR DESCRIPTION
**Pull request by handorff**

### CODE REVIEW

File: apps/api_web/test/api_web/controllers/trip_controller_test.exs [hunk 1]
```diff
@@ -133,6 +133,17 @@ defmodule ApiWeb.TripControllerTest do
       assert index_data(conn, %{"route" => "2", "direction_id" => "0"}) == []
     end
 
+    test "filters by name", %{conn: conn} do
+      trip = %Model.Trip{id: "1", name: "name"}
+      :ok = State.Trip.new_state([trip])
+
+      [^trip] = State.Trip.filter_by(%{names: ["name"]})
+      [] = State.Trip.filter_by(%{names: ["not_a_name"]})
```
Comment (paulswartz) [inline apps/api_web/test/api_web/controllers/trip_controller_test.exs]: these should be tests in `apps/state`

File: apps/api_web/test/api_web/controllers/trip_controller_test.exs [hunk 2]
```diff
@@ -133,6 +133,17 @@ defmodule ApiWeb.TripControllerTest do
...
+
+      assert index_data(conn, %{"name" => "name"}) == [trip]
+      assert index_data(conn, %{"name" => "not_a_name"}) == []
```
Comment (paulswartz) [inline apps/api_web/test/api_web/controllers/trip_controller_test.exs]: there should also be a test that an empty name returns an error.

Comment (paulswartz) [reply apps/api_web/test/api_web/controllers/trip_controller_test.exs]: I think `index_data(conn, %{"name" => ","})` will return trips, instead of an error.
